shimmingtoolbox/shim/sequencer.py

```diff
@@ -281,12 +280,15 @@ def _cal_shimmed_anat_orient(coefs, coils, nii_mask_anat, nii_fieldmap, slices,
                                      order=1,
                                      mode='grid-constant',
                                      cval=0).get_fdata()
-
-    shimmed_anat_orient = np.zeros_like(fieldmap_anat)
-    for i_shim in range(len(slices)):
-        corr = np.sum(coefs[i_shim] * coils_anat, axis=3, keepdims=False)
-        shimmed_anat_orient[..., slices[i_shim]] = fieldmap_anat[..., slices[i_shim]] + corr[..., slices[i_shim]]
-
+    shimmed_anat_orient = fieldmap_anat
+    nb_channel = np.shape(coefs)[1]
+    dimx = np.shape(coils_anat)[0]
+    dimy = np.shape(coils_anat)[1]
+    for i_shim in list_shim_slice:
+        # We want to do the np.sum with a 2D matrix to get it aster
+        coils_anat_reduced = np.reshape(coils_anat[:, :, slices[i_shim], :], (-1, nb_channel))
+        corr = np.sum(coefs[i_shim] * coils_anat_reduced, axis=1, keepdims=False)
+        shimmed_anat_orient[..., slices[i_shim]] += np.reshape(corr, (dimx, dimy, 1))
```

Comment:
I think this can crash because the slice dimension is not always 1, I suggest using the way I had showed you.